File: backend/src/main/java/com/bookbook/domain/rent/entity/Rent.java
package com.bookbook.domain.rent.entity;
import com.bookbook.global.jpa.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

// 25.08.04 현준
@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Rent extends BaseEntity {

    private Long lenderUserId; // 대여자 ID
    private String title; // 글 제목
    private String bookCondition; // 책 상태
    private String bookImage; // 글쓴이가 올린 책 이미지 URL
    private String address; // 사용자 주소
    private String contents; // 대여 내용

    @Enumerated(EnumType.STRING) // enum 이름을 String으로 DB에 저장하도록 지정
    private RentStatus rentStatus; // 대여 상태(Available, Loaned, Finished, Deleted)

    // 알라딘 API로 받아온 책 관련 속성
    private String bookTitle; // 책 제목
    private String author; // 책 저자
    private String publisher; // 책 출판사
    private String category; // 책 카테고리
    private String description; // 책 설명
}

File: backend/src/main/java/com/bookbook/domain/rent/entity/Rent.kt
package com.bookbook.domain.rent.entity

import com.bookbook.global.jpa.entity.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated

// 25.08.29 현준
// 대여 게시글을 나타내는 엔티티 클래스
@Entity
class Rent(
    // 대여자 정보
    @Column(nullable = false)
    var lenderUserId: Long, // 대여자 ID

    // 게시글 기본 정보
    @Column(nullable = false)
    var title: String, // 글 제목
    
    @Column(nullable = false)
    var bookCondition: String, // 책 상태
    
    @Column(nullable = false)
    var bookImage: String, // 글쓴이가 올린 책 이미지 URL
    
    @Column(nullable = false)
    var address: String, // 사용자 주소
    
    @Column(nullable = false, length = 500)
    var contents: String, // 대여 내용

    // 대여 상태
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var rentStatus: RentStatus, // 대여 상태(Available, Loaned, Finished, Deleted)

    // 알라딘 API 도서 정보
    @Column(nullable = false)
    var bookTitle: String, // 책 제목
    
    @Column(nullable = false)
    var author: String, // 책 저자
    
    @Column(nullable = false)
    var publisher: String, // 책 출판사
    
    @Column(nullable = false)
    var category: String, // 책 카테고리
    
    @Column(nullable = false, length = 500)
    var description: String // 책 설명
) : BaseEntity()